breakerbp/failure_ratio.go

```diff
@@ -12,6 +12,12 @@ import (
 	"github.com/reddit/baseplate.go/metricsbp"
 )
 
+// CircuitBreaker is the interface that baseplate expects a circuit breaker to implement.
+type CircuitBreaker interface {
+	// Execute should wrap the given function call in circuit breaker logic and return the result.
+	Execute(func() (interface{}, error)) (interface{}, error)
+}
```

Comment:
nit: maybe move it to its own file (e.g. `interface.go`) so it makes more sense if we are adding other implementations in the future.